lib/jsdom/living/nodes/HTMLInputElement-impl.js

```diff
@@ -340,7 +340,7 @@ class HTMLInputElementImpl extends HTMLElementImpl {
     switch (valueAttributeMode(this.type)) {
       // https://html.spec.whatwg.org/multipage/input.html#dom-input-value-value
       case "value":
-        return this._value !== null ? this._value : "";
+        return this._getValue();
```

Comment:
@domenic Interestingly it appears that the accessor for the input already handles this case and [has test coverage](https://github.com/jsdom/jsdom/blob/master/test/to-port-to-wpts/htmlinputelement.js#L10-L12). Therefore in order to maintain the existing API I've pushed the logic back to `_getValue()` and have called that method from here. 

That said, I do wonder if `_getValue()` should be removed and the `FormData` logic changed to be:

```js
formDataSet.push({ name, value: field.value, type });
``` 

I am not familiar with this project so am likely missing a lot of context but a project search showed that `_getValue` was implemented in:

- [`HTMLButtonElement`](https://github.com/jsdom/jsdom/blob/master/lib/jsdom/living/nodes/HTMLButtonElement-impl.js#L31-L34)
- [`HTMLTextareaElement`](https://github.com/jsdom/jsdom/blob/master/lib/jsdom/living/nodes/HTMLTextAreaElement-impl.js#L37-L40) (which oddly _also_ has a getter for [`value()`](https://github.com/jsdom/jsdom/blob/master/lib/jsdom/living/nodes/HTMLTextAreaElement-impl.js#L66-L68) which calls [`_getAPIValue `](https://github.com/jsdom/jsdom/blob/ecb22a27627ace4e3d5db09cabf566ec606c36ad/lib/jsdom/living/nodes/HTMLTextAreaElement-impl.js#L33-L35)
- [`HTMLInputElement`](https://github.com/jsdom/jsdom/blob/ecb22a27627ace4e3d5db09cabf566ec606c36ad/lib/jsdom/living/nodes/HTMLInputElement-impl.js#L182-L184) (this file).

It appears to only called by the [`FormData`](https://github.com/jsdom/jsdom/blob/master/lib/jsdom/living/xhr/FormData-impl.js#L149) logic and I have a suspicion the cases it is handling are already handled by `value` accessor.

1. While `HTMLButtonElement` defines `getValue()`, the loop in `FormData` [will bail out if the element is a button](https://github.com/jsdom/jsdom/blob/master/lib/jsdom/living/xhr/FormData-impl.js#L108-L110).
1. The logic for [handling checkboxes/radio](https://github.com/jsdom/jsdom/blob/master/lib/jsdom/living/xhr/FormData-impl.js#L137-L139) is already handled in this switch block with the `default` and `default/on` cases.
1. The `HTMLTextAreaElement` implementation of `_getValue()` already calls `normalizeToCRLF`. This is likely the reason for the [`entry.type !== "textarea"`](https://github.com/jsdom/jsdom/blob/master/lib/jsdom/living/xhr/FormData-impl.js#L162) guard later in this method.

What do you think?

Reply:
@domenic Is there another way I could approach this or are you happy with the changes in ad1415d?

Reply:
Hmm, I think this is confusing, indeed. Let me try playing around with something more like the spec...